### DISCUSSION TITLE
Setting text to an icon over mqtt doesnt work

### DISCUSSION THREAD

**Discussion by RossiRPI** [HASwitchPlate/openHASP#135] [General]
Setting text field of a label to an icon over mqtt doesn't work. Seems like the received MQTT value gets treated as a string and doesn't get checked for a special character/icon. 
For example sending hasp/test_plate/command/p0b1.text="\uf743 Rain" sets the text field to "uf743 R" instead of the icon for rain and the word Rain.
Thought I'd post this here before creating a bug report.

**Comment by fvanroie** [comment]
`\uf743` formatting only works inside the `json` or `jsonl` commands. e.g. `json ["p0b1.text=\uf743 Rain"]`
When using other commands like `p0b1.text` you have to UTF-8 encode the *raw* payload yourself.
You don't need quotes around the text parameter.

**Reply by RossiRPI** [reply]
OK... so is it possible to change an icon over MQTT? Could you give me an example of how to do it? I know that I can send json commands over serial and also use json to configure the screens.

**Reply by fvanroie** [reply]
serial and mqtt commands are equivalent: e.g.
`hasp/test_plate/command/json` `["p0b1.text=\uf743 Rain"]`

[Documentation](https://haswitchplate.github.io/openHASP-docs/#fonts/#custom-fonts):

> To use an icon in a text you need to prefix the UTF-8 value with `\u`. To ensure proper decoding the payload should be used with a `json` or `jsonl` command.
> 
> `jsonl` example:
> 
> `{"obj":"label","id":10,"x":0,"y":0,"w":70,"h":50,"parentid":5,"text":"\uf00c OK"}`
> 
> `json` example:
> 
> `["p2b10.text=\uf00c OK"]`
